pkg/metrics/metrics.go

```diff
@@ -0,0 +1,79 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package metrics creates and registers metrics objects with Prometheus
+// and sets the Prometheus HTTP handler for /metrics
+package metrics
+
+import (
+	"net/http"
+	"sync"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+var registerMetrics sync.Once
+
+const (
+	catalogNamespace = "servicecatalog" // Prometheus namespace (nothing to do with k8s namespace)
```

Comment:
Would this allow for you to monitor many service catalogs from the same prometheus ? Or would the metrics get mushed together?

Reply:
Prometheus can easily be configured to monitor multiple Catalog Controllers all at the same time.   Additionally Prometheus adds a 'kubernetes_pod_name' attribute to every metric that would distinguish the different controller's metrics from each other.  If you graph the same metric name from multiple controllers, each will display separately unless you sum them or apply some other function that consolidates the values down to a single metric.